### PR TITLE
[risk=no] refactor cohort review detail state

### PR DESCRIPTION
**Pull request by panentheos**
This changes the state management on the review detail page to be more react-friendly. It will make gradual conversion of this page much more straightforward. Highlights:
* Several resolvers were removed, and the page component now does its own data loading.
* Most state now formally lives on the page component. Child components receive the data as `Input` bindings, as well as callback functions to request updates, as appropriate.

### CODE REVIEW

File: ui/src/app/cohort-review/annotation-item/annotation-item.component.ts [hunk 1]
```diff
@@ -74,7 +71,7 @@ export class AnnotationItemComponent implements OnInit, OnChanges, AfterContentC
     private cdref: ChangeDetectorRef
   ) {}
 
-  ngOnChanges() {
+  ngOnChanges(changes) {
```
Comment (blrubenstein) [inline ui/src/app/cohort-review/annotation-item/annotation-item.component.ts]: Does this need to be added here?

File: ui/src/app/cohort-review/annotation-list/annotation-list.component.html [hunk 2]
```diff
@@ -6,16 +6,15 @@ <h4 class="annotations-color">Annotations</h4>
     (click)="openManager()">
     <clr-icon shape="plus-circle" size="21" title="Manage Annotations"></clr-icon>
   </button>
-  <button *ngIf="(annotations$ | async).length> 0"
+  <button *ngIf="annotationList.length> 0"
```
Comment (blrubenstein) [inline ui/src/app/cohort-review/annotation-list/annotation-list.component.html]: Nit: Add a space before `>`

File: ui/src/app/cohort-review/detail-all-events/detail-all-events.component.html [hunk 3]
```diff
@@ -18,8 +18,7 @@
     </clr-dg-filter>
   </clr-dg-column>
 
-  <clr-dg-row *clrDgItems="let datum of data"
-    (click)="onClick(datum)">
```
Comment (blrubenstein) [inline ui/src/app/cohort-review/detail-all-events/detail-all-events.component.html]: Why was this removed? Just out of curiosity/not understanding.

File: ui/src/app/cohort-review/detail-page/detail-page.ts [hunk 4]
```diff
@@ -12,19 +15,55 @@ export class DetailPage implements OnInit, OnDestroy {
...
   sidebarOpen = true;
   participant: Participant;
-  subscription: Subscription;
-  participantId: number;
-  constructor(private route: ActivatedRoute) {}
+  annotations: ParticipantCohortAnnotation[];
+  annotationDefinitions: CohortAnnotationDefinition[];
+  subscriptions: Subscription[] = [];
+  constructor(
+    private reviewAPI: CohortReviewService,
```
Comment (blrubenstein) [inline ui/src/app/cohort-review/detail-page/detail-page.ts]: I would name these `reviewService` and `defsService` because we have a distinction between service and API. (Service is generated for angular, and API is generated for fetch)

File: ui/src/app/cohort-review/annotation-item/annotation-item.component.ts (hunk 1 above)
Comment (panentheos) [reply ui/src/app/cohort-review/annotation-item/annotation-item.component.ts]: no, that was left over from development

File: ui/src/app/cohort-review/detail-all-events/detail-all-events.component.html (hunk 3 above)
Comment (panentheos) [reply ui/src/app/cohort-review/detail-all-events/detail-all-events.component.html]: The downstream event was not actually used.

File: ui/src/app/cohort-review/detail-page/detail-page.ts (hunk 4 above)
Comment (panentheos) [reply ui/src/app/cohort-review/detail-page/detail-page.ts]: I agree with the reasoning, but there is an established pattern of calling these 'API' in the angular code, so I'm inclined to keep that consistency to make the refactoring smoother.

File: ui/src/app/cohort-review/detail-tabs/detail-tabs.component.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <clr-tabs>
   <clr-tab>
-    <button clrTabLink id="detail-summary-tab"(click) = "getDomainsParticipantsData()">Summary</button>
+    <button clrTabLink id="detail-summary-tab">Summary</button>
```
Comment (smazumd) [inline ui/src/app/cohort-review/detail-tabs/detail-tabs.component.html]: when we are switching between other tabs summary tab clicking is not smooth it takes time to open the tab. Not sure what is causing the issue.

File: ui/src/app/cohort-review/detail-page/detail-page.html [hunk 6]
```diff
@@ -1,33 +1,41 @@
-<div class="detail-page" [class.sidebar-open]="sidebarOpen">
+<ng-container *ngIf="participant && annotations && annotationDefinitions">
```
Comment (smazumd) [inline ui/src/app/cohort-review/detail-page/detail-page.html]: I observed if I click on participant from participants page, it is routing to the clicked participant and no spinner comes up, and it waits to get all data and then charts spinners start spinning. there is a time gap between getting chart data and all other data(like participants id and all). So I think we need to add a spinner here before ng-container , where we need to check ' <div class="spinner spinner-inverse" *ngIf="!participant && !annotations && !annotationDefinitions">Loading...</div>' -- something like this. but may be there will be more better way  to handle it. it was an observation.

Comment (panentheos) [reply ui/src/app/cohort-review/detail-page/detail-page.html]: Good call, I added a spinner.

File: ui/src/app/cohort-review/detail-tabs/detail-tabs.component.html (hunk 5 above)
Comment (panentheos) [reply ui/src/app/cohort-review/detail-tabs/detail-tabs.component.html]: I noticed this as well on certain participants. I think there are some issues with the data manipulation in the charts, and it ends up doing substantial extra work when there are many data points, and this leads to a noticeable pause on render. I think this is already happening without this PR, but it's masked because we reload the data every time, so the initial render just shows the spinner. We should investigate that separately.

File: ui/src/app/cohort-review/detail-page/detail-page.html (hunk 6 above)
Comment (smazumd) [reply ui/src/app/cohort-review/detail-page/detail-page.html]: nit : I think we need some css so that spinner will be in the center of the page. you can add this below css-
.review-spinner {
  position: absolute;
  top: 45%;
  left: 50%;
}
and add this class to the spinner div.
